velox/functions/prestosql/Slice.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/type/Type.h"
+
+namespace facebook::velox::functions {
+
+/// @tparam IndexKind The `TypeKind` of the function `Slice` start and length
+/// type.
+template <TypeKind IndexKind>
+void registerSliceFunction(const std::string& prefix);
+
+extern template void registerSliceFunction<TypeKind::BIGINT>(
+    const std::string& prefix);
+
+extern template void registerSliceFunction<TypeKind::INTEGER>(
+    const std::string& prefix);
```

Comment:
Do we need to define `registerSliceFunction<TypeKind::BIGINT>` and `registerSliceFunction<TypeKind::INTEGER>`. I assume the template function registerSliceFunction should be enough.

Reply:
If I'm understanding correctly, define it here can reduce the binary size and compile time.

https://www.sandordargo.com/blog/2023/11/08/extern-templates